integration-tests/chopsticks/src/tests/xcm/types.ts

```diff
@@ -0,0 +1,48 @@
+import type { ApiPromise } from '@polkadot/api'
+import type { SubmittableExtrinsic } from '@polkadot/api/types'
+import type { KeyringPair } from '@polkadot/keyring/types'
+import type { SetupOption, EventFilter } from '@acala-network/chopsticks-testing'
+
+export interface Storage {
```

Comment:
Ok so this assumes the tests always include sender, relay, and receiver? What if we want to test XCM hopping on two different chains? What if we don't care about the receiver storage? I think a simpler and more generic option would be to simply use a `Record<string, Record<string, Record<string, unknown>>>`, so basically replacing your first-class properties with a key, which you can set and get in each test case. In that case, this whole type could be just a type alias like `export type Storage = Record<string, Record<string, Record<string, unknown>>>`. You can of course refactor each nested type into its own for more clarity, if you wish.

Reply:
If you want, you can then `export const storageKey = <string>` for each chain config.